lib/ansible/plugins/lookup/secret_server.py

```diff
@@ -0,0 +1,226 @@
+            type: string
+            env:
+                - name: SS_DOMAIN_ANSIBLE
+            required: false (required only with SAAS)
+        address:
+            description:
+                - FQDN / IP for the Secret Server.
+            default: None
+            type: string
+            env:
+                - name: SS_ADDRESS_ANSIBLE
+            required: true (either ENV or kwargs)
+        key:
+            description:
+                - Name of the key for which you want to obtain the value.
+            default: password
+            type: string
+            required: false
+'''
+
+EXAMPLES = '''
+    - name: "Get passowrd for customer_1 with ENV variables"
+      debug:
+        msg: "{{ lookup('secret_server', 'customer_1') }}"
+
+    - name: "Get Key=Username for customer_1 with ENV variables"
+      debug:
+        msg: "{{ lookup('secret_server', 'customer_1', key='username') }}"
+
+    - name: "Get passowrd for customer_1"
+      debug:
+        msg: >
+            "{{ lookup('secret_server', 'customer_1',
+            address='server1.example.com', domain='example.com',
+            username='awesome_user', password='awesome_password') }}"
+'''
+
+RETURN = '''
+    raw:
+        description:
+            - List of values associated with input keys.
+        type: list
+        elements: strings
+'''
+
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+import os
+
+try:
+    import requests
+    import urllib3
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+# Check if ENV variables are present
+if os.getenv('SS_USER_ANSIBLE') is not None:
+    SS_USER_ANSIBLE = os.getenv('SS_USER_ANSIBLE')
+else:
+    SS_USER_ANSIBLE = None
+
+if os.getenv('SS_PASSWORD_ANSIBLE') is not None:
+    SS_PASSWORD_ANSIBLE = os.getenv('SS_PASSWORD_ANSIBLE')
+else:
+    SS_PASSWORD_ANSIBLE = None
+
+if os.getenv('SS_DOMAIN_ANSIBLE') is not None:
+    SS_DOMAIN_ANSIBLE = os.getenv('SS_DOMAIN_ANSIBLE')
+else:
+    SS_DOMAIN_ANSIBLE = None
+
+if os.getenv('SS_ADDRESS_ANSIBLE') is not None:
+    SS_ADDRESS_ANSIBLE = os.getenv('SS_ADDRESS_ANSIBLE')
+else:
+    SS_ADDRESS_ANSIBLE = None
+
+
+class SecretServer(object):
+    def __init__(self, username=None, password=None, secret_key=None,
+                 domain=None, address=None, key='password'):
+        self.username = username
+        self.password = password
+        self.secret_key = secret_key
+        self.domain = domain
+        self.address = address
+        self.key = key
+
+    def connect_api(self, path, address, auth_key=None, method='GET',
+                    connection='secure', headers=None, body=None):
+
+        if connection.upper() == 'SECURE':
+            connection = 'https://'
+        else:
+            connection = 'http://'
+
+        api_url = "{}{}" . format(connection, address)
+        url = "{}{}" . format(api_url, path)
```

Comment:
This is not py2.6 compatible.  You will need to index these.

Reply:
When you say index do you mean this:

```
api_url = "{0}{1}" . format(connection, address)
url = "{0}{1}" . format(api_url, path)
```